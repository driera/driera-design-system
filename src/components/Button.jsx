import React from 'react'

const Button =  () => {
  return (
    <button>dds button</button>
  )
};

export default Button;
