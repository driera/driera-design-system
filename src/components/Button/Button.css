button {
  margin: 0;
  outline: none;
  border: none;
  color: var(--dark-color);
}

.button--main {
  min-width: 30px;
  padding: .5em 2.8em;
  margin: .5em;
  border-radius: 5px;
  border: 1px solid var(--primary-color);
  font-size: 12px;
  font-family: var(--secondary-font);
  font-weight: bold;
  color: var(--light-color);
  background: transparent;
  cursor: pointer;
  transition: 0.3s ease-in;
}

.button--main:hover {
  background: var(--primary-color);
}

.button--square {
  padding: .5em;
}

.button--big {
  font-size: 16px;
}

.button--alert {
  color: var(--alert-color);
  border-color: var(--alert-color);
}

.button--alert:hover {
  color: var(--dark-color);
  background: var(--alert-color);
}

.button--warning {
  color: var(--warning-color);
  border-color: var(--warning-color);
}

.button--warning:hover {
  color: var(--dark-color);
  background: var(--warning-color);
}
