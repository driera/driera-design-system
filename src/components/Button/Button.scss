button {
  margin: 0;
  outline: none;
  border: none;
  color: var(--dark-color);
}

.button--main {
  min-width: 30px;
  padding: .5em 2.8em;
  margin: .5em;
  border-radius: 5px;
  border: 1px solid var(--primary-color);
  font-size: 12px;
  font-family: var(--secondary-font);
  font-weight: bold;
  color: var(--light-color);
  background: transparent;
  cursor: pointer;
  transition: 0.3s ease-in;
  
  &:hover {
    background: var(--primary-color);
  }
  
  &.v-square {
    padding: .5em;
  }
  
  &.sz-big {
    font-size: 16px;
  }
  
  &.st-alert {
    color: var(--alert-color);
    border-color: var(--alert-color);
    
    &:hover {
      color: var(--dark-color);
      background: var(--alert-color);
    }
  }
  
  &.st-warning {
    color: var(--warning-color);
    border-color: var(--warning-color);
    
    &:hover {
      color: var(--dark-color);
      background: var(--warning-color);
    }
  }
}

