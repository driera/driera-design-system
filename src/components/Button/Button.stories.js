import React from 'react'
import { storiesOf } from '@storybook/react';
import Button from './Button';
import { action } from '@storybook/addon-actions';
import { withKnobs, select, text, radios } from '@storybook/addon-knobs';
import '../../styles/styles.scss';

storiesOf('Button', module)
  .addDecorator(withKnobs)
  .add('square', () => (
      <>
      <Button text="Regular button" onClick={action('Regular button action click')} />
      <Button text="-" ratio="square" onClick={action('square button action click')} />
      <Button text="+" ratio="square" onClick={action('square button action click')} />
      <Button text="3D" ratio="square" onClick={action('square button action click')} />
      </>
  ))
  .add('Size', () => (
    <>
    <Button text="Regular button" onClick={action('Regular button action click')} />
    <Button text="BIG button" size="big" onClick={action('BIG button action click')} />
    </>
  ))
  .add('Color', () => (
    <>
      <Button text="Regular Button" onClick={action('Regular button action click')} />
      <Button text="Warning Button" color="warning" onClick={action('Warning button action click')} />
      <Button text="Alert Button" color="alert" onClick={action('Alert button action click')} />
    </>
  ))
  .add('Customizable', () => (
    <Button 
      text={text(
        'Text',
        `Knob Button`
      )} 
      color={select(
        'Select color',
        ['primary', 'warning', 'alert'], 
        'primary'
      )}
      size={radios(
        'Size', 
        {
          Regular: '', 
          Big: 'big'
        }, 
        ''
      )}
      ratio={radios(
        'Ratio', 
        {
          Regular: '',
          Square: 'square'
        },
        ''
      )}
    />
  ))
