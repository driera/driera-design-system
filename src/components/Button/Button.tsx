import React from 'react'
import './Button.css';

type ButtonProps = {
  text: string;
  ratio?: string;
  size?: string;
  color?: string;
  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
}

const Button = ({text, ratio, size, color, ...props} : ButtonProps) : JSX.Element => {
  const ratioClass = ratio && ratio.length > 0 ? `button--${ratio}` : '';
  const sizeClass = size && size.length > 0 ? `button--${size}` : '';
  let colorClass = color && color.length > 0 ? `button--${color}` : '';
  
  const classCollection = `button--main ${sizeClass} ${ratioClass} ${colorClass}`;

  return (
    <button className={classCollection} {...props}>
      {text}
    </button>
  )
}

export default Button;
