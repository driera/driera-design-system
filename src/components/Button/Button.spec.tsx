import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "./Button";

const clickCallback = jest.fn()

describe("Button", () => {
  it("Can be clicked", () => {
    const text = "Click me!";
    const { getByText, container } = render(
      <Button text={text} onClick={(clickCallback)} />
    );

    const button = getByText(text);
    fireEvent.click(button);

    expect(container).toMatchSnapshot()
    expect(clickCallback).toBeCalled()
  });

  it("Can be big", () => {
    const text = "Click me!";
    const { container } = render(
      <Button text={text} onClick={(clickCallback)} size="big" />
    );

    expect(container).toMatchSnapshot()
  });  
});
