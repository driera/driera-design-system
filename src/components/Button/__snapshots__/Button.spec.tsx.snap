// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Can be big 1`] = `
<div>
  <button
    class="button--main button--big  "
  >
    Click me!
  </button>
</div>
`;

exports[`Button Can be clicked 1`] = `
<div>
  <button
    class="button--main   "
  >
    Click me!
  </button>
</div>
`;
